app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionDetailsFragment.kt

```diff
@@ -0,0 +1,104 @@
+package org.wikipedia.suggestededits
+
+import android.os.Bundle
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.GONE
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import kotlinx.android.synthetic.main.fragment_contribution_diff_detail.*
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.history.HistoryEntry
+import org.wikipedia.json.GsonUnmarshaller
+import org.wikipedia.page.PageActivity
+import org.wikipedia.page.PageTitle
+import org.wikipedia.suggestededits.SuggestedEditsContributionDetailsActivity.Companion.EXTRA_SOURCE_CONTRIBUTION
+import org.wikipedia.suggestededits.SuggestedEditsContributionsFragment.Contribution.Companion.EDIT_TYPE_ARTICLE_DESCRIPTION
+import org.wikipedia.suggestededits.SuggestedEditsContributionsFragment.Contribution.Companion.EDIT_TYPE_IMAGE_CAPTION
+import org.wikipedia.util.DateUtil
+import org.wikipedia.util.GradientUtil
+import org.wikipedia.views.ViewUtil
+
+class SuggestedEditsContributionDetailsFragment : Fragment() {
+    var contribution: SuggestedEditsContributionsFragment.Contribution? = null
+
+    companion object {
+        fun newInstance(): SuggestedEditsContributionDetailsFragment {
+            return SuggestedEditsContributionDetailsFragment()
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        super.onCreateView(inflater, container, savedInstanceState)
+        return inflater.inflate(R.layout.fragment_contribution_diff_detail, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        topView.background = GradientUtil.getPowerGradient(R.color.green90, Gravity.TOP)
+        back.setOnClickListener { requireActivity().onBackPressed() }
+        contribution = GsonUnmarshaller.unmarshal(SuggestedEditsContributionsFragment.Contribution::class.java, requireActivity().intent.getStringExtra(EXTRA_SOURCE_CONTRIBUTION))
+        setUpContributionDetails()
+    }
+
+    private fun setUpContributionDetails() {
+        contributionContainer.setOnClickListener { startTypeSpecificActivity() }
+        contributionDiffText.text = getString(R.string.suggested_edits_contribution_label, contribution!!.sizeDiff)
+        contributionDetailText.text = contribution!!.description
+        revisionText.text = contribution!!.revisionId.toString()
+
+        contributionTitle.text = contribution!!.title
+        if (contribution!!.imageUrl.isEmpty() || contribution!!.imageUrl.equals("null")) contributionImage.visibility = GONE else ViewUtil.loadImageWithRoundedCorners(contributionImage, contribution!!.imageUrl)
```

Comment:
Please change the `.equals("null")` to `== "null"`. Just curious, does the `imageUrl` return a string "null"? or a `null`?

Reply:
it returns a string "null"! very weird.